contracts/examples/ExampleSlidingWindowOracle.sol

```diff
@@ -17,44 +17,55 @@ contract ExampleSlidingWindowOracle {
     using SafeMath for uint;
 
     struct Observation {
-        uint32 blockTimestamp;
+        uint timestamp;
         uint price0Cumulative;
         uint price1Cumulative;
     }
 
     address public immutable factory;
-    // the desired amount of time over which the moving average should be computed
-    // may not be exceed max uint32, as block timestamps cannot be compared for larger windows
-    uint32 public immutable windowSize;
-    // the number of observations stored for each pair.
-    // as granularity increases from 1, more frequent updates are needed, but moving averages become more precise
+    // the desired amount of time over which the moving average should be computed, e.g. 24 hours
+    uint public immutable windowSize;
+    // the number of observations stored for each pair, i.e. how many price observations are stored for the window.
+    // as granularity increases from 1, more frequent updates are needed, but moving averages become more precise.
     // averages are computed over intervals with sizes in the range:
     //   [windowSize - (windowSize / granularity) * 2, windowSize]
     // e.g. if the window size is 24 hours, and the granularity is 24, the oracle will return the average price for
     //   the period:
     //   [now - [22 hours, 24 hours], now]
     uint8 public immutable granularity;
+    // this is redundant with granularity and windowSize, but stored for gas savings & informational purposes.
+    uint public immutable periodSize;
 
     // mapping from pair address to a list of price observations of that pair
     mapping(address => Observation[]) public pairObservations;
 
-    constructor(address factory_, uint32 windowSize_, uint8 granularity_) public {
+    constructor(address factory_, uint windowSize_, uint8 granularity_) public {
         require(granularity_ > 1, 'SlidingWindowOracle: GRANULARITY');
-        require((windowSize_ / granularity_) * granularity_ == windowSize_, 'SlidingWindowOracle: UNEVEN_EPOCHS');
+        require(
+            (periodSize = windowSize_ / granularity_) * granularity_ == windowSize_,
+            'SlidingWindowOracle: WINDOW_NOT_EVENLY_DIVISIBLE'
+        );
         factory = factory_;
         windowSize = windowSize_;
         granularity = granularity_;
     }
 
+    // returns the index of the observation corresponding to the given timestamp
+    function observationIndexOf(uint timestamp) public view returns (uint8 index) {
+        uint epochPeriod = timestamp / periodSize;
+        return uint8(epochPeriod % granularity);
```

Comment:
personally i still find

```
return uint8((timestamp % windowSize) / periodSize);
```

more intuitive

Reply:
I see things like epochSeconds and epochMilliseconds in a lot of code and I think epochPeriod makes intuitive sense. Breaking it up into two lines just feels easier for me to reason about, I got confused by the one liner before

Reply:
kk, it's not a merge blocker